GitHubReleases.ps1

```diff
@@ -193,8 +201,993 @@ function Get-GitHubRelease
         'AccessToken' = $AccessToken
         'TelemetryEventName' = $MyInvocation.MyCommand.Name
         'TelemetryProperties' = $telemetryProperties
-        'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -BoundParameters $PSBoundParameters -Name NoStatus -ConfigValueName DefaultNoStatus)
+        'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    $multipleResult = Invoke-GHRestMethodMultipleResult @params
+
+    # Add additional property to ease pipelining
+    foreach ($result in $multipleResult)
+    {
+        Add-Member -InputObject $result -Name 'ReleaseId' -Value $result.id -MemberType NoteProperty -Force
+    }
+
+    return $multipleResult
+}
+
+filter New-GitHubRelease
```

Comment:
Given the discussion in https://github.com/microsoft/PowerShellForGitHub/pull/228, it did occur to me that perhaps this could be `Publish-GitHubRelease`.

But that all depends on your philosophy and approach, like:

- If you want to match PowerShell command verbs with REST API verbs
- If you want a minimal set of verbs you feel simplifies use of the module
- If you're looking for more PowerShell-oriented verb usage that seeks to abstract the underlying API call a bit more

There's also the degree to which `New-*` returns a value, compared to `Publish-*`.

So here I think `New-*` applies reasonably well to `GitHubRelease`, since the release doesn't exist yet, even though the command also makes it available for others (being similar to `Publish-Module`). Then for `New-GitHubReleaseAsset`, the question is whether the asset is being created (is the asset the file, or GitHub's concept of it), or just published to the existing release.

Anyway, I only raise this because of the discussion in #228. Despite having opinions on verbs, I generally think that (1) I have less concept of verbs since I use fewer modules than most PowerShell users and (2) a lot of command verb discussions (and naming discussions generally) get very protracted without being very helpful, so I try to live and let live in that respect.

Reply:
I love the thinking here and appreciate the thoughts and feedback.  I think I'll leave this as `New` for now, and can revisit possibly adding an alias based on additional feedback later on.